lib/jbuilder/collection_renderer.rb

```diff
@@ -0,0 +1,118 @@
+require 'delegate'
+require 'active_support/concern'
+
+begin
+  require 'action_view/renderer/collection_renderer'
+rescue LoadError
+  require 'action_view/renderer/partial_renderer'
+end
+
+class Jbuilder
+  module CollectionRenderable # :nodoc:
+    extend ActiveSupport::Concern
+
+    class_methods do
+      def supports_collection_rendering?
+        superclass.private_method_defined?(:build_rendered_template) && superclass.private_method_defined?(:build_rendered_collection)
+      end
+    end
```

Comment:
These sound like they would be costly to run on each collection render, could we extract them with something like this?

```suggestion
    class_methods do
      supported = superclass.private_method_defined?(:build_rendered_template) && superclass.private_method_defined?(:build_rendered_collection)
      define_method(:enabled?) { supported }
    end
```